### PR TITLE
Fix problem with index_name being unavailable for a job

### PR DESCRIPTION
**Pull request by artem-shelkovnikov**
Fixes a problem reported in Slack:

```
[FMWK][15:05:17][CRITICAL] 'index_name'
Traceback (most recent call last):
  File "/Users/artemshelkovnikov/git_tree/connectors-py/connectors/services/job_cleanup.py", line 64, in _process_orphaned_jobs
    job.index_name is not None
  File "/Users/artemshelkovnikov/git_tree/connectors-py/connectors/byoc.py", line 208, in index_name
    return self.doc_source["_source"]["connector"]["index_name"]
KeyError: 'index_name'
```

Problem happens because when job is created the property "source.index_name" is never populated, while logic that cleans up jobs uses this key.

This fix is suboptimal - better refactor the whole structure of classes, but I know @wangch079 is addressing some of it in his PRs, thus I'm proposing this temporary fix for the problem until refactoring lands.

#### Pre-Review Checklist
- [x] Covered the changes with automated tests
- [x] Tested the changes locally by running `make test` and `make ftest NAME=mysql`
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

File: connectors/byoc.py [hunk 1]
```diff
@@ -207,7 +207,11 @@ def duration(self):
     def index_name(self):
         return self.doc_source["_source"]["connector"]["index_name"]
 
-    async def start(self, trigger_method=JobTriggerMethod.SCHEDULED, filtering=None):
+    async def start(self, connector, trigger_method=JobTriggerMethod.SCHEDULED, filtering=None):
+        # Okay that's temporary here. When we separate scheduling from execution , this code will go away
```
Comment (tarekziade) [inline connectors/byoc.py:211]: I would add `XXX` or `TODO` as a prefix to this comment

**Review by tarekziade — APPROVED** [review]

**Review by wangch079 — APPROVED** [review]
This will be eventually fixed after #460 and #474 are merged. But let's have a working `main` first.

File: connectors/byoc.py (hunk 1 above)
Comment (timgrein) [reply connectors/byoc.py:211]: ++ for "# TODO:", which is caught by nearly every IDE and shows up in the corresponding TODOs tab.

**Comment by artem-shelkovnikov** [comment]
Closing as not needed any more

**Pull request closed without merging**